crates/wdl-engine/src/cache/hash.rs

```diff
@@ -188,16 +188,19 @@ impl Hashable for Digest {
 
 impl Hashable for EvaluationPath {
     fn hash(&self, hasher: &mut Hasher) {
-        match self {
-            Self::Local(path) => {
-                PathKind::Local.hash(hasher);
-                path.hash(hasher);
-            }
-            Self::Remote(url) => {
-                PathKind::Remote.hash(hasher);
-                url.hash(hasher);
-            }
+        if let Some(path) = self.as_local() {
```

Comment:
Personally, I like things like this to be `match`es unless there is a good reason to switch. I can't think of one here, but perhaps you have one. If, for no other reason, (a) it's implied that the `match` is exhaustive and [avoiding the `unreachable!`] and (b) it forces you to update the code if you add a new variant. That being said, I don't have a strong preference necessarily.